test/inductor/test_triton_kernels.py

```diff
@@ -3656,6 +3658,185 @@ def add(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
             )
             add_compiled(x, y).mean()
 
+    @requires_gpu
+    @common_utils.parametrize(
+        "backend", ["non-strict", "eager", "aot_eager", "inductor"]
+    )
+    @common_utils.parametrize("with_perf_model", [True, False])
+    def test_triton_kernel_prune_configs_by(self, backend, with_perf_model):
+        # for non-strict mode
+        libname = "my_cool_namespace"
+        opname = "my_triton_operator"
+
+        records = {}
+
+        def early_config_prune(configs, named_args, **kwargs):
+            # we need to save the records to the returned config
+            records["run_early_config_prune"] = True
+            if "N" in kwargs and kwargs["N"] == 1024:
+                records["capture_kwargs"] = True
+            # named args are: dst, src, add_float
+            if "dst" in named_args and "src" in named_args and len(named_args) == 3:
+                records["capture_named_args"] = True
+            return [configs[0]]
+
+        # this will pick the Config with the largest block size  (128)
+
+        def perf_model(*args, **kwargs):
+            records["run_perf_model"] = True
+            return kwargs["BLOCK_SIZE"] * -1
+
+        if with_perf_model:
+            prune_configs_by = {"perf_model": perf_model, "top_k": 1}
+        else:
+            prune_configs_by = {"early_config_prune": early_config_prune}
+
+        @triton.autotune(
+            configs=[
+                triton.Config(kwargs={"BLOCK_SIZE": 32}),
+                triton.Config(kwargs={"BLOCK_SIZE": 128}),
+            ],
+            key=["N"],
+            prune_configs_by=prune_configs_by,
+        )
+        @triton.jit
+        def prune_by_kernel(
+            dst,
+            src,
+            add_float,
+            N,
+            BLOCK_SIZE: tl.constexpr,
+        ):
+            offsets = tl.program_id(0) * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+            x = tl.load(src + offsets, mask=offsets < N)
+            # Let's make sure we always select a block size of 128 based on our perf_model
+            if BLOCK_SIZE == 128:
+                x = x + add_float
+            tl.store(dst + offsets, x, mask=offsets < N)
+
+        if backend == "non-strict":
+            decorator = torch.library.triton_op(f"{libname}::{opname}", mutates_args={})
```

Comment:
I think, `non-strict` should be orthogonal to the `backend`. I.e., `non-strict` determines how the Triton kernel is wrapped, whereas the backend is passed to `torch.compile`. When `non-strict` is set, we should wrap the kernel into `triton_op`. And then we should `torch.compile` in any case: whether `non-strict` is enabled or not.